File: Data Structures & Algorithms/jump-game-ii/submission-1.java
class Solution {
    public int jump(int[] nums) {
        int curr = 0;
        int jumps = 0;
        while (curr < nums.length - 1)
        {
            int reach = nums[curr] + curr; // 2
            int idx = curr + 1; // 3
            if (reach >= nums.length - 1 )
            {
                jumps++;
                break;
            }
            for (int i = curr + 2; i <= reach; i++) // 4
            {
                if (nums[i] + i > nums[idx] + idx) // 
                { 
                    idx = i;
                }
            }
            curr = idx;
            jumps++;
        }
        return jumps;
    }
}

File: Data Structures & Algorithms/word-break/submission-1.cpp
class Solution {
public:
    bool wordBreak(string s, vector<string>& wordDict) {
        map<string, int> hashMap;
        set<int> idxs;
        for (string str : wordDict)
        {
            hashMap[str] = 1;
        }
        idxs.insert(0);
        for (int i =1; i < s.size(); i++)
        {
            for (int idx : idxs)
            {
                if (hashMap.contains(s.substr(idx, i - idx)))
                    idxs.insert(i);
            }
        }
        for (int idx : idxs)
        {
            if(hashMap.contains(s.substr(idx)))
                return true;
        }
        return false;

    }

    /*bool stringDfs(string curr, int idx,string s, map<string, int> hashMap)
    {
        if (idx == s.size())
            return curr == "";
        
        curr += s[idx];
        idx++;
        bool res;
        if (hashMap.contains(curr))
        {
            res = stringDfs("", idx, s, hashMap);
        }
        return res || stringDfs(curr, idx, s, hashMap);
        
    }*/
};
